src/platform/plugins/shared/controls/public/controls/esql_control/get_esql_control_factory.tsx

```diff
@@ -0,0 +1,156 @@
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { BehaviorSubject } from 'rxjs';
+import { EuiComboBox } from '@elastic/eui';
+import { useBatchedPublishingSubjects, PublishingSubject } from '@kbn/presentation-publishing';
+import { ESQLVariableType } from '@kbn/esql-validation-autocomplete';
+import { esqlVariablesService } from '@kbn/esql-variables/common';
+import { ESQL_CONTROL } from '../../../common';
+import type { ESQLControlState, ESQLControlApi } from './types';
+import { ControlFactory } from '../types';
+import { uiActionsService } from '../../services/kibana_services';
+import { initializeDefaultControlApi } from '../initialize_default_control_api';
+import { initializeESQLControlSelections } from './esql_control_selections';
+
+const displayName = i18n.translate('controls.esqlValuesControl.displayName', {
+  defaultMessage: 'Static values list',
+});
+
+export const getESQLControlFactory = (): ControlFactory<ESQLControlState, ESQLControlApi> => {
+  return {
+    type: ESQL_CONTROL,
+    order: 3,
+    getIconType: () => 'editorChecklist',
+    getDisplayName: () => displayName,
+    buildControl: async (initialState, buildApi, uuid, controlGroupApi) => {
+      const selectedOptions$ = new BehaviorSubject<string[]>(initialState.selectedOptions ?? []);
+      const variableName$ = new BehaviorSubject<string>(initialState.variableName ?? '');
+      const variableType$ = new BehaviorSubject<ESQLVariableType>(
+        initialState.variableType ?? ESQLVariableType.VALUES
+      );
+      // initialize the variable
+      esqlVariablesService.addVariable({
+        key: initialState.variableName,
+        value: initialState.selectedOptions[0],
+        type: initialState.variableType,
+      });
+      const hasSelections$ = new BehaviorSubject<boolean>(
+        Boolean(initialState.selectedOptions?.length)
+      );
+      const defaultControl = initializeDefaultControlApi({ ...initialState });
+
+      const selections = initializeESQLControlSelections(initialState);
+
+      const onSaveControlCb = (updatedState: ESQLControlState) => {
+        controlGroupApi?.replacePanel(uuid, {
+          panelType: 'esqlControl',
+          initialState: updatedState,
+        });
+      };
+
+      const api = buildApi(
+        {
+          ...defaultControl.api,
+          defaultPanelTitle: new BehaviorSubject<string | undefined>(initialState.title),
+          isEditingEnabled: () => true,
+          getTypeDisplayName: () => displayName,
+          onEdit: async () => {
```

Comment:
Editing an ES|QL control doesn't update the header of the flyout - it still says "Create ES|QL control": 

![image](https://github.com/user-attachments/assets/f32c7736-deb8-49b9-906e-9739762b477d)

Reply:
Done https://github.com/elastic/kibana/pull/202875/commits/61ee6f4e6e711b52751882e43805c01bc91d60f4